app/models/host/managed.rb

```diff
@@ -537,11 +537,29 @@ def jumpstart?
     operatingsystem.family == "Solaris" and architecture.name =~/Sparc/i rescue false
   end
 
+  HOSTGROUP_INHERITED_ATTRIBUTES = %w{puppet_proxy_id puppet_ca_proxy_id environment_id compute_profile_id}
+
+  def apply_inherited_attributes(attributes)
+    new_hostgroup_id = attributes['hostgroup_id']
+    return attributes if new_hostgroup_id.nil?
+
+    hostgroup = self.hostgroup || Hostgroup.find(new_hostgroup_id)
+    hostgroup = Hostgroup.find(new_hostgroup_id) unless hostgroup.id == new_hostgroup_id
+    return attributes unless hostgroup
```

Comment:
if I understand these 3 lines correctly, the return can't be ever reached, hostgroup is either already present or is found by id. In first case, hostgroup is always true, in second either true (we found it) or ActiveRecord::RecordNotFound exception is raised, if you didn't intend to raise exceptions, use `Hostgroup.find_by_id`

but I think these lines could be changed, we don't need to build new hostgroup variable and reimplement active record reloading
```ruby
self.hostgroup_id = new_hostgroup_id # this ensures self.hostgroup is reloaded if needed
# now you can use self.hostgroup.send(...)
```